File: MyProject/Source/MyProject/Private/BasicFrame/Libs/EventHandle/AddOnceAndCallOnceEventDispatch.cpp
#include "MyProject.h"
#include "AddOnceAndCallOnceEventDispatch.h"

void AddOnceAndCallOnceEventDispatch::addEventHandle(EventDispatchDelegate handle)
{
	if (!this->isExistEventHandle(handle))
	{
		Super::addEventHandle(handle);
	}
}

void AddOnceAndCallOnceEventDispatch::dispatchEvent(IDispatchObject* dispatchObject)
{
	Super::dispatchEvent(dispatchObject);
	this->clearEventHandle();
}

File: Project/MyProject/Tool/CSToolKit/Project/ToolKit/ToolKit/Source/Libs/EventHandle/AddOnceAndCallOnceEventDispatch.cs
namespace SDK.Lib
{
    public class AddOnceAndCallOnceEventDispatch : EventDispatch
    {
        override public void addEventHandle(ICalleeObject pThis, MEventDispatchAction<IDispatchObject> handle, uint eventId = 0)
        {
            if (!this.isExistEventHandle(pThis, handle, eventId))
            {
                base.addEventHandle(pThis, handle, eventId);
            }
        }

        override public void dispatchEvent(IDispatchObject dispatchObject)
        {
            base.dispatchEvent(dispatchObject);

            this.clearEventHandle();
        }
    }
}